### PR TITLE
Refactor browser files, improve consistency and add functionality

### PR DESCRIPTION
**Pull request by david-tejada**
The goal of this PR is to reduce duplication, add consistency and complete functionality where possible. It should close issue #532.

Things done:

- Move most of the common browser functionality to browser.py and browser_mac.py
- Implement tab actions in those same two files and in the specific browser apps when they differ.
- Change mac shortcuts for tab next and tab previous to `ctrl-tab` and `ctrl-shift-tab` that have less issues with international keyboards. (Maybe a bit out of the scope of this PR)
- Implement action `browser.focus_page` and add command to trigger it.
- Implement action `user.tab_duplicate` for all the browsers.
- Implement commands `bookmarks sidebar` and `history sidebar` and their respective actions for Firefox.
- Fix some shortcuts that were incorrect.
- Raise NotImplementedError for the actions that couldn't be implemented with a clear message of the reason.
- Add support for Opera browser.
- Add .talon files for all the browsers. This makes it easier if the user wants to add custom commands.
- Move all the tag activation to .talon files
- Remove commented out commands in browser.talon. Many of those can now be performed with Rango.

I know this is a big PR. I just wanted to implement browser.focus_page and I got a little caught up 😅. I have manually tested all the functionality for all the browsers for Mac, Windows and Linux.

### CODE REVIEW

**Review by splondike — APPROVED** [review]
I like the idea, and the implementation looks good to me. I've only given it a bit of a test on FF/Chrome on Linux.

Will leave this another week or so for others to have a chance to look at and then if you fix up the merge conflict we can put it in :D.

File: core/windows_and_tabs/tabs.py [hunk 1]
```diff
@@ -1,7 +1,9 @@
-from talon import Module, actions
+from talon import Module, actions, app
 
 mod = Module()
 
+cmd_ctrl = "cmd" if app.platform == "mac" else "ctrl"
```
[the remark this answers is not in the document]
Comment (david-tejada) [reply core/windows_and_tabs/tabs.py]: I see, I left it there after I reverted some changes. I removed it

**Comment by david-tejada** [comment]
Nice! Merge conflict resolved

**Comment by david-tejada** [comment]
I have added Vivaldi. I have also checked that it works in all the platforms

**Comment by splondike** [comment]
OK, it's done! Thanks for all the work, and let's hope nothing slipped through :D.

**Comment by david-tejada** [comment]
Thanks! I think you can close #532 now

**Pull request merged**